library/ssl_tls.c

```diff
@@ -3744,9 +3744,16 @@ static int ssl_tls13_session_save(const mbedtls_ssl_session *session,
     size_t hostname_len = (session->hostname == NULL) ?
                           0 : strlen(session->hostname) + 1;
 #endif
+
+#if defined(MBEDTLS_SSL_SRV_C) && \
+    defined(MBEDTLS_SSL_EARLY_DATA) && defined(MBEDTLS_SSL_ALPN)
+    const uint8_t alpn_len = (session->ticket_alpn == NULL) ?
+                             0 : (uint8_t) strlen(session->ticket_alpn);
```

Comment:
It is a pity that we cannot use `mbedtls_ssl_session_set_ticket_alpn()` in `_load()`. Proposition: we use two bytes to store the ALPN length. Thus here it would be:
```suggestion
    size_t alpn_len = (session->ticket_alpn == NULL) ?
                             0 : strlen(session->ticket_alpn) + 1;
```
line 3778:
```
needed += 2 /* alpn_len */
```
line 3834
```
        MBEDTLS_PUT_UINT16_BE(alpn_len, p, 0);                              
        p += 2;  
```
...
line 4232
```
opaque ticket_alpn<0..256>;
```
That way it would also be aligned with how it is done for the host name.